src/handlers/check_commits/canonicalize_issue_links.rs

```diff
@@ -0,0 +1,56 @@
+use crate::{config::CanonicalizeIssueLinksConfig, github::GithubCommit};
+
+pub(super) fn issue_links_in_commits(
+    _conf: &CanonicalizeIssueLinksConfig,
+    full_repo_name: &str,
+    commits: &[GithubCommit],
+) -> Option<String> {
+    let uncanonicalized_issue_links_commits = commits
+        .into_iter()
+        .filter(|c| super::super::canonicalize_issue_links::LINKED_RE.is_match(&c.commit.message))
+        .map(|c| format!("    - {}\n", c.sha))
+        .collect::<String>();
+
+    if uncanonicalized_issue_links_commits.is_empty() {
+        None
+    } else {
+        Some(format!(
+            r"There are un-canonicalized issue links (such as `#123`) in the commit messages of the following commits.
+  *Please canonicalize them by prefixing them with `{full_repo_name}#` (such as `{full_repo_name}#123`).*
```

Comment:
Contrary to https://github.com/rust-lang/triagebot/issues/1924, I'm proposing to canonicalize them, since I can see valid reasons to link issues in individual commits.

Reply:
Hum, but your reasoning about avoiding those spamming references to commits in issues is compelling.

Reply:
Canonicalized is better than just raw issue links, but they can still spam, yeah.

Reply:
Issue links being spammy in commits feels IMO being better reason anyway to avoid them in commits than being un-canonicalized. It also seems easier to justify to contributor to do the work to remove them.

Reply:
Switch to warn against every issue links and ask to remove them.

Reply:
I think that if you want to protect against issue links in PR descriptions, you will most likely also want to do it in commit messages. So I would ideally like to unify both and rename them, but I guess that we don't have a good story for atomic renames of triagebot options, without doing some deprecation period or aliases or something.

So let's add a new feature. Maybe `[detect-commit-issue-links]`? But what you proposed is also fine, so up to you.

Reply:
> but I guess that we don't have a good story for atomic renames of triagebot options

No need to atomic update, but can just add an alias, so that's what I did, the new handler `issue-links` aliases the old one `canonicalize-issue-links`.